packages/flutter_markdown/lib/src/markdown.dart

```diff
@@ -0,0 +1,472 @@
+      h5: h5 != null ? h5 : this.h5,
+      h6: h6 != null ? h6 : this.h6,
+      em: em != null ? em : this.em,
+      strong: strong != null ? strong : this.strong,
+      blockquote: blockquote != null ? blockquote : this.blockquote,
+      blockSpacing: blockSpacing != null ? blockSpacing : this.blockSpacing,
+      listIndent: listIndent != null ? listIndent : this.listIndent,
+      blockquotePadding: blockquotePadding != null ? blockquotePadding : this.blockquotePadding,
+      blockquoteDecoration: blockquoteDecoration != null ? blockquoteDecoration : this.blockquoteDecoration,
+      codeblockPadding: codeblockPadding != null ? codeblockPadding : this.codeblockPadding,
+      codeblockDecoration: codeblockDecoration != null ? codeblockDecoration : this.codeblockDecoration
+    );
+  }
+
+  final TextStyle a;
+  final TextStyle p;
+  final TextStyle code;
+  final TextStyle h1;
+  final TextStyle h2;
+  final TextStyle h3;
+  final TextStyle h4;
+  final TextStyle h5;
+  final TextStyle h6;
+  final TextStyle em;
+  final TextStyle strong;
+  final TextStyle blockquote;
+  final double blockSpacing;
+  final double listIndent;
+  final double blockquotePadding;
+  final BoxDecoration blockquoteDecoration;
+  final double codeblockPadding;
+  final BoxDecoration codeblockDecoration;
+
+  Map<String, TextStyle> _styles;
+
+  void _init() {
+    _styles = {
+      'a': a,
+      'p': p,
+      'li': p,
+      'code': code,
+      'pre': p,
+      'h1': h1,
+      'h2': h2,
+      'h3': h3,
+      'h4': h4,
+      'h5': h5,
+      'h6': h6,
+      'em': em,
+      'strong': strong,
+      'blockquote': blockquote
+    };
+  }
+}
+
+class Markdown extends StatefulComponent {
+  Markdown({
+    this.data,
+    this.markdownStyle,
+    this.syntaxHighlighter
+  });
+
+  final String data;
+  final MarkdownStyle markdownStyle;
+  final SyntaxHighlighter syntaxHighlighter;
+
+  State<Markdown> createState() => new _MarkdownState();
+}
+
+class _MarkdownState extends State<Markdown> {
+
+  void initState() {
+    super.initState();
+
+    MarkdownStyle markdownStyle = config.markdownStyle;
+    if (markdownStyle == null)
+      markdownStyle = new MarkdownStyle.defaultFromTheme(Theme.of(context));
+
+    SyntaxHighlighter syntaxHighlighter = config.syntaxHighlighter;
+    if (syntaxHighlighter == null)
+      syntaxHighlighter = new DartSyntaxHighlighter();
+
+    _cachedBlocks = _blocksFromMarkup(config.data, markdownStyle, syntaxHighlighter);
+  }
+
+  List<_Block> _cachedBlocks;
+
+  Widget build(BuildContext context) {
+    List<Widget> blocks = <Widget>[];
+    for (_Block block in _cachedBlocks) {
+      blocks.add(block.build(context));
+    }
+
+    return new Column(
+      alignItems: FlexAlignItems.stretch,
+      children: blocks
+    );
+  }
+}
+
+List<_Block> _blocksFromMarkup(String data, MarkdownStyle markdownStyle, SyntaxHighlighter syntaxHighlighter) {
+  var lines = data.replaceAll('\r\n', '\n').split('\n');
```

Comment:
no var.

Also, this is a relatively expensive operation (multiple string copies). It would be better to just support `\r` directly in the tokeniser.